src/physics/em/PhotoelectricInteractor.i.hh

```diff
@@ -0,0 +1,190 @@
+    --shell_id;
+
+    // Construct interaction for change to primary (incident) particle
+    Interaction result = Interaction::from_absorption();
+
+    // If the binding energy of the sampled shell is greater than the incident
+    // photon energy, no secondaries are produced and the energy is deposited
+    // locally.
+    if (el.shells[shell_id].binding_energy > inc_energy_)
+    {
+        return result;
+    }
+
+    // Outgoing secondary is an electron
+    result.secondaries    = {photoelectron, 1};
+    photoelectron->def_id = shared_.electron_id;
+
+    // Electron kinetic energy is the difference between the incident photon
+    // energy and the binding energy of the shell
+    photoelectron->energy = MevEnergy{
+        inc_energy_.value() - el.shells[shell_id].binding_energy.value()};
+
+    // Direction of the emitted photoelectron is sampled from the
+    // Sauter-Gavrila distribution
+    photoelectron->direction = this->sample_direction(rng);
+
+    // TODO: Atomic relaxation
+
+    return result;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Sample a direction according to the Sauter-Gavrila distribution.
+ *
+ * \note The Sauter-Gavrila distribution for the K-shell is used to sample the
+ * polar angle of a photoelectron. This performs the same sampling routine as
+ * in Geant4's G4SauterGavrilaAngularDistribution class, as documented in
+ * section 6.3.2 of the Geant4 Physics Reference (release 10.6) and section
+ * 2.1.1.1 of the Penelope 2014 manual.
+ */
+template<class Engine>
+CELER_FUNCTION Real3 PhotoelectricInteractor::sample_direction(Engine& rng) const
+{
+    constexpr MevEnergy min_energy{1.e-6};
+    constexpr MevEnergy max_energy{100.};
+    real_type           energy_per_mecsq;
+
+    if (inc_energy_ > max_energy)
+    {
+        // If the incident gamma energy is above 100 MeV, use the incident
+        // gamma direction for the direction of the emitted photoelectron.
+        return inc_direction_;
+    }
+    else if (inc_energy_ < min_energy)
+    {
+        // If the incident energy is below 1 keV, set it to 1 keV.
+        energy_per_mecsq = min_energy.value() * shared_.inv_electron_mass;
+    }
+    else
+    {
+        energy_per_mecsq = inc_energy_.value() * shared_.inv_electron_mass;
+    }
+
+    // Calculate Lorentz factors of the photoelectron
+    real_type gamma = energy_per_mecsq + 1.;
+    real_type beta  = std::sqrt(energy_per_mecsq * (gamma + 1.)) / gamma;
+    real_type a     = (1. - beta) / beta;
+
+    // Second term inside the brackets in Eq. 2.8 in the Penelope manual
+    real_type b = 0.5 * beta * gamma * energy_per_mecsq * (gamma - 2.);
+
+    // Maximum of the rejection function g(1 - cos \theta) given in Eq. 2.8,
+    // which is attained when 1 - cos \theta) = 0
+    real_type g_max = 2. * (1. / a + b);
+
+    // Rejection loop: sample 1 - cos \theta
+    real_type g;
+    real_type one_minus_costheta;
```

Comment:
Maybe use "nu" to match the penelope manual?

Reply:
yep that's better 😉